src/languages/stan.js

```diff
@@ -512,41 +410,74 @@ export default function(hljs) {
       hljs.HASH_COMMENT_MODE,
       BLOCK_COMMENT,
       {
-        // hack: in range constraints, lower must follow "<"
-        begin: /<\s*lower\s*=/,
-        keywords: 'lower'
+        scope: 'symbol',
+        match: /\s(pi|e|sqrt2|log2|log10)(?=\()/,
+        relevance: 0
+      },
+      {
+        match: regex.concat(/[<,]\s*/, regex.either(...RANGE_CONSTRAINTS), /\s*=/),
+        keywords: RANGE_CONSTRAINTS
       },
       {
-        // hack: in range constraints, upper must follow either , or <
-        // <lower = ..., upper = ...> or <upper = ...>
-        begin: /[<,]\s*upper\s*=/,
-        keywords: 'upper'
+        scope: 'keyword',
+        match: /\btarget\s*/,
       },
       {
-        className: 'keyword',
-        begin: /\btarget\s*\+=/
+        // highlights the 'T' in T[,] for only Stan language distributrions
+        match: [
+          /~\s*/,
+          regex.either(...DISTRIBUTIONS),
+          /(?:\(\))/,
+          /\s*T(?=\s*\[)/
+        ],
+        scope: {
+          2: "built_in",
+          4: "keyword"
+        }
       },
       {
-        begin: '~\\s*(' + hljs.IDENT_RE + ')\\s*\\(',
-        keywords: DISTRIBUTIONS
+        // highlights distributrions that end with special endings
+        scope: 'built_in',
+        match: /\w+(_lpdf|_lupdf|_lpmf|_cdf|_lcdf|_lccdf|_qf)(?=\s*[\(.*\)])/
```

Comment:
Are we sure `(?=\s*[\(.*\)])` this is correct?

What are we trying to match here again?  The `.*` inside of [] looks very strange.

Reply:
they must only be highlighted if there is a parenthesis and I want it to be closed with any character or space in between

Reply:
Also if we always highlight these (regardless of prefix), can all of these suffixes just be removed from the keyword lists entirely since we have this rule?

Reply:
I've kind of half-way allowed user defined functions getting highlighted. This only allows user defined distributions which must have one of those endings. I wanted to make sure that if someone inputs something that looks like a function with `()` but hasn't defined it, it doesn't get highlighted and make them think it's a valid function. It seems much more likely that if the function has one of those endings that it is already defined in the function block. 

Do we allow this? Or should we only allow highlighting of things defined explicitly in the language? I'm ok with including the `keywords: DISTRIBUTIONS` here and not highlighting user defined distributions.

Reply:
> If someone inputs something that looks like a function with () but hasn't defined it

We highlight *snippets* (often), not full code blocks.  Plus we have no way of knowing if someone "defined" something earlier or not since we don't keep track of any context like that. Unless we detect it as it's being defined, but then later we can't detect a "reference to something defined in the past"...

I feel like this should perhaps be all or nothing.  My thinking:

- distributions and other built in functions are `built_in`
- all function dispatch `doSomething(`, ie `[validFuncName][lookahead("(")]`) are `title.function`.

Is a `identifier(` guaranteed to be a function dispatch in Stan?

Reply:
I added all udfs as title.function and I broke out distributions so they are only used with `built_ins`. I'm digging what this looks like.